src/pages/Settings/Settings.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react'
+import useForm from 'react-hook-form'
+import * as yup from 'yup'
+import Button from 'components/Button'
+
+import './Settings.module.css'
+
+export const PORT_NUMBER_REGEX = /^\d{4,6}$/
+export const EMAIL_REGEX = /^\S+@\S+$/i
+
+const SettingsValidationSchema = yup.object().shape({
+  hostName: yup.string().required(),
+  hostPubKey: yup.string().required(),
+  registrationEmail: yup.string()
+    .email()
+    .required(),
+  deviceName: yup.string().required(),
+  networkId: yup.string().required(),
+  deviceAdminPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hcAdminPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hcNetworkPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hostingPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required()
+})
+
+export function Settings ({
+  settings,
+  updateSettings,
+  factoryReset,
+  toggleSshAccess,
+  history: { push }
+}) {
+  // if (props.loading) return <h4>Loading Settings</h4>
+
+  const goToMenu = () => push('/menu')
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: settings,
+    validationSchema: SettingsValidationSchema
+  })
+
+  const [sshAccessVal, setSshAccess] = useState(false)
+
+  const handleViewTos = (e) => {
+    e.preventDefault()
```

Comment:
I haven't been doing this. What does it achieve?

Reply:
Keeps a form from HTML submitting (page reloading). May or may not be necessary anymore, carry over from Lisa code.

Reply:
👍